ls1intum/Artemis
# `General`: Fix an issue with a few buttons that were not correctly disabled

### Checklist
#### General


#### Client

### Motivation and Context
This pull request closes #4199. 
On the programming exercise creation page, the generate button was clickable even when it was disabled. 

### Description
This issue was caused by the fact that the jhi-button listened to the "click" event instead of the "onClick" event.
While (click) will always be called when the user clicks on the button (onClick) will not be called when the button is disabled.
I therefore replaced the (click) event handlers for all jhi-buttons with (onClick) event handlers, as (onClick) should always be preferred when using jhi-buttons. This only changes the behavior when the jhi-button is disabled and not when it is enabled.

### Steps for Testing

Prerequisites:
- 1 Editor/Instructor

1. Log in to Artemis
2.  Open the programming exercise creation dialogue
3. Do not insert any input to any field
4. Navigate to the bottom of the page and make sure the Generate button is disabled
5. Click the Generate button 
6. No action should happen

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
Unchanged

### Screenshots
No visual changes.
